### DISCUSSION TITLE
Programming Failed

### DISCUSSION THREAD

**Discussion by guatepapi4** [forkineye/ESPixelStick#507] [Q&A]
Attempted flashing my 2 new V3 sticks last night. Both gave me failed programming errors. I was able to program one to blink the LED using the Arduino Cloud Editor. Any thoughts on what next steps to try? The desktop I used is minimum10 years old and running Win10 (Very Slowly). New laptop comes in to weeks and I anticipate the issue is just the age of the machine given that I can communicate and program using the IDE.

**Comment by MartinMueller2003** [comment]
A little bit more information is needed here. Did you build the image yourself or are you using a prebuilt image? What tool did you use to upload the image? If you are using an IDE, you should be using vscode with platformio. It has what you need to upload. If you are using the prebuilt images you need to use the ESPFlashTool from Forkineye.

**Reply by guatepapi4** [reply]
Using the prebuilt image. Stable Release 3.2
I used the ESPFlashtool. All attempts failed on two sticks. I was able to see the com ports so believe the cable was correct as it recognized the sticks as CH340 devices.
When that failed I attempted to program using the Arduino Cloud IDE to just download something to it as I started above. Did this to confirm device was good. I was able to program using the Cloud IDE. Now trying to figure out why the flashtool doesnt download.

**Comment by forkineye** [comment]
How exactly is it failing? Can post a screenshot of the output window or any error dialogs?

**Comment by guatepapi4** [comment]
I’ll get a screenshot later but all I get is a little bit of gibberish in
the serial wi Dow when I first plug in. There is a single line of parameter
info with the baud rate and nothing after that. Like it never even starts
the download.

**Comment by guatepapi4** [comment]
![image](https://user-images.githubusercontent.com/103146807/162189628-2d813036-80dc-4a8a-aba5-f62556d28205.png)

**Reply by forkineye** [reply]
We need to see what's in the Serial Output section.

**Reply by guatepapi4** [reply]
Boy is this Electrical Engineer feeling dumb. Last time it was mostly squares with a few other random characters. Nothing that tells you anything comprehendible. I'll try to pull it again tonight. Thanks for the assistance!

**Reply by MartinMueller2003** [reply]
No Worries. That gobeldygook at the start is the ESP sending out information at a different buadrate than what we generally use. Once the software boots, it switches to the expected baudrate and we can read the output. What we need is to see the gobldygook to see if the device entered the programming state at the right time.

When looking into an issue, there is no such thing as too much information. It is best to send us everything you see so that we can see where things may have gone wrong. For instance, we only see a little bit of the output in your screenshot. But it would be better if you scrolled back and copied ALL of the text in both windows into this request.

Martin

**Reply by guatepapi4** [reply]
The serial window is blank when I click upload. I get the following when I push reset on wemos:

![image](https://user-images.githubusercontent.com/103146807/162340108-0735ce47-b839-476e-b8d6-a2dcbf6b39b1.png)

**Reply by forkineye** [reply]
That's very odd. Which version of the firmware are you trying to flash?  3.2 is the latest stable.  Also, do you have any other software running that may try accessing the serial ports like 3D printer slicers?  Cura is known to cause issues blocking the serial ports, but the error you get is usually different than what you're seeing.

**Reply by forkineye** [reply]
Also, we switched to the python based flash tool (same that's included with Arduino) in 3.2.  You could try 3.1 as a test which uses the older binary flash tool on the backend.

**Reply by guatepapi4** [reply]
Flashed 3.2 It's a very old machine so nothing would surprise me. No other software running that would use com ports. The port I used only appeared after I plugged in the Wemos. Getting ready to try the V4 with logs and 3.1 shortly.

**Comment by MartinMueller2003** [comment]
Here is a newer version of the Flashtool. This version supports only V4 but it has the ability to save a log of all messages. Due to git restrictions, I cannot upload a zip file so the file has a txt stuck on the end of it. Remove the .txt and then unzip the file. After that, use it like normal. After you try to program the device, click the save logs button, give the log a name. Then attach the log file to this conversation
[ESPixelStickV4Dist.zip.txt](https://github.com/forkineye/ESPixelStick/files/8447969/ESPixelStickV4Dist.zip.txt)
.

**Comment by guatepapi4** [comment]
Interesting happenings. The V4 tool failed. Logs attached from attempts at both devices I have.

The 3.1 tool was able to successfully download. I only downloaded one of the two in case you would like to do some additional troubleshooting from the blank state.
[Download Log 2.txt](https://github.com/forkineye/ESPixelStick/files/8453797/Download.Log.2.txt)
[Download Log.txt](https://github.com/forkineye/ESPixelStick/files/8453798/Download.Log.txt)

**Reply by forkineye** [reply]
Sounds like there's an issue with the Python version of the flash tool on your system for some reason. What version of Windows is it?

**Reply by guatepapi4** [reply]
Win10 Home 10.0.19042 Build 19042

![image](https://user-images.githubusercontent.com/103146807/162485612-f6666678-9e1e-4575-a700-67f9c50bffeb.png)
